File: Easy/real_fake.r
test.cases <- c("9999 9999 9999 9999",
				"9999 9999 9999 9993")

#args <- commandArgs(trailingOnly=TRUE)
#test.cases <- strsplit(readLines(args[[1]], warn=FALSE), '\n')
cat(sapply(test.cases, function(test) {
	if (length(test) > 0) {
		card <- gsub(" ", "", test)
		card <- strsplit(card, split = "")[[1]]
		card <- as.numeric(card)
		double_nums <- sum(card[seq(1,16,2)] * 2)
		single_nums <- sum(card[seq(2,16,2)])
		Valid <- double_nums + single_nums
		if (Valid %% 10 == 0) {
			"Real"
		} else {
			"Fake"
		}
	}
}), sep = "\n")

File: Easy/real_fake.py
data = ["9999 9999 9999 9999",
		"9999 9999 9999 9993"]

#import sys
#test_cases = open(sys.argv[1], 'r')
#data = test_cases.readlines()
for test in data:
	test = test.rstrip()
	card = test.replace(" ", "")
	card = map(int, list(card))
	SUM = 0
	for i in range(0, 16):
		if i %2 == 0:
			SUM = SUM + (card[i]*2)
		else:
			SUM = SUM + card[i]
	if SUM % 10 == 0:
		print "Real"
	else:
		print "Fake"
	
#test_cases.close()